pkgs/development/ruby-modules/gem-config/default.nix

```diff
@@ -447,9 +457,20 @@ in
     buildInputs = lib.optionals stdenv.isDarwin [ CoreServices ];
   };
 
+  hpricot = attrs: {
+    dontBuild = false;
+    patches = [
+      # Fix incompatible function pointer conversion errors with clang 16
+      ./hpricot-fix-incompatible-function-pointer-conversion.patch
```

Comment:
is there an upstream patch that we can fetch via fetchpatch?

Reply:
Upstream hasn’t been updated in over a decade and recommends using nokogiri.

Reply:
There’s a PR with a fix, but it touched a _lot_ of stuff unnecessarily.

Reply:
ah, its _why's hpricot, maybe we should remove it

Reply:
It’s a dependency of ejson, so that would probably need to be updated.

(Does ejson even use Ruby anymore? It appears to be written in Go now.)

Reply:
looks like published gem is just binaries  https://my.diffend.io/gems/ejson/1.3.1/1.4.1

so yes its written in go now :-) :cactus:

Reply:
ah, they use ronn to build manual and it depends on hpricot,